avocado/utils/linux_modules.py

```diff
@@ -208,34 +208,91 @@ def get_loaded_modules():
         return [astring.to_text(_.split(b' ', 1)[0]) for _ in proc_modules]
 
 
-def check_kernel_config(config_name):
+def check_kernel_config(config_name, session=None):
     """
     Reports the configuration of $config_name of the current kernel
 
     :param config_name: Name of kernel config to search
+    :param session: guest session, command is run on host if None
     :type config_name: str
     :return: Config status in running kernel (NOT_SET, BUILTIN, MODULE)
     :rtype: :class:`ModuleConfig`
     """
+    if session is None:
+        kernel_version = platform.uname()[2]
+        config_file = '/boot/config-' + kernel_version
+        # Sometimes a kernel config's name may contain another kernel config
+        # E.g.
+        # CONFIG_PREEMPT_VOLUNTARY=y
+        # CONFIG_PREEMPT is not set
+        # So we need to search CONFIG_XXX= in kconfig file
+        config_info = process.run('grep {}= {}'.format(config_name,
+                                  config_file), shell=True)\
+            .stdout_text.strip()
+    else:
+        kernel_version = session.cmd('uname -r').strip()
+        config_file = '/boot/config-' + kernel_version
+        config_info = session.cmd('grep {}= {}'.format(config_name,
+                                  config_file)).strip()
+
+    line = config_info.split('=')
```

Comment:
If `grep` finds more than one match on a kconfig, for whatever reason (such as a commented-out entry), this will cause problems.  Example:

```
...
CONFIG_CC_VERSION_TEXT="gcc (GCC) 11.2.1 20220127 (Red Hat 11.2.1-9)"
CONFIG_CC_IS_GCC=y
# CONFIG_CC_IS_GCC=n
...
```

`line` will become: `'CONFIG_CC_IS_GCC=y\n# CONFIG_CC_IS_GCC=n'` and then the split will return 3 items, causing this function to errouneously return `ModuleConfig.NOT_SET`.

The previous code considered only the first match... at least that behavior needs to remain IMO.

Reply:
> If `grep` finds more than one match on a kconfig, for whatever reason (such as a commented-out entry), this will cause problems. Example:
> 
> ```
> ...
> CONFIG_CC_VERSION_TEXT="gcc (GCC) 11.2.1 20220127 (Red Hat 11.2.1-9)"
> CONFIG_CC_IS_GCC=y
> # CONFIG_CC_IS_GCC=n
> ...
> ```
> 
> `line` will become: `'CONFIG_CC_IS_GCC=y\n# CONFIG_CC_IS_GCC=n'` and then the split will return 3 items, causing this function to errouneously return `ModuleConfig.NOT_SET`.
> 
> The previous code considered only the first match... at least that behavior needs to remain IMO.

I think `grep ^CONFIG_CC_IS_GCC= $config_file`  could avoid this situation.